react/GalleryWrapper.js

```diff
@@ -1,81 +1,42 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import QueryString from 'query-string'
 import { graphql } from 'react-apollo'
+import { ExtensionPoint } from 'render'
 
-import WrappedSpinner from './components/WrappedSpinner'
-import productsQuery from './queries/productsQuery.gql'
+export default class GalleryWrapper extends Component {
+  constructor(props) {
+    super(props)
+    this.state = this.getSearchParamsFromUrl()
+  }
 
-import { ExtensionPoint } from 'render'
+  getSearchParamsFromUrl() {
+    let query = window.location && window.location.pathname.slice(1)
+    const queryParams = window.location && QueryString.parse(window.location.search)
+    if (queryParams && queryParams.Q) {
+      query = [query].concat(queryParams.Q.split(',')).join('/')
+    }
+    return {
+      query,
+      map: queryParams && queryParams.map,
+      orderBy: queryParams && queryParams.O,
+      page: parseInt((queryParams && queryParams.page) || 1),
+    }
+  }
 
-class GalleryWrapper extends Component {
+  componentWillReceiveProps() {
+    this.setState(this.getSearchParamsFromUrl())
```

Comment:
Maybe move this function (`getSearchParamsFromUrl`) to the new [`static getDerivedStateFromProps`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops)